src/Controller/UserController.php

```diff
@@ -0,0 +1,130 @@
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Security\Http\Authentication\UserAuthenticatorInterface;
+use App\Security\LoginAuthenticator;
+
+
+class UserController extends AbstractController
+{
+    private $passwordHasher;
+
+    public function __construct(UserPasswordHasherInterface $passwordHasher)
+    {
+        $this->passwordHasher = $passwordHasher;
+    }
+
+    #[Route('/add/user', name: 'add_user')]
+    public function addUser(Request $request, UserPasswordHasherInterface $userPasswordHasher, UserAuthenticatorInterface $userAuthenticator, LoginAuthenticator $authenticator, EntityManagerInterface $entityManager): Response
+    {
+        $user = new User();
+        $form = $this->createForm(UserType::class, $user);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            // encode the plain password
+            $user->setPassword(
+                $userPasswordHasher->hashPassword(
+                    $user,
+                    $form->get('password')->getData()
+                )
+            );
+
+            $entityManager->persist($user);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('app_admin');
+
+            
+
+        
+            // do anything else you need here, like send an email
+        }
+        return $this->render('admin/add.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    private function createDeleteConfirmationForm(): \Symfony\Component\Form\FormInterface
+    {
+        return $this->createFormBuilder()
+            ->add('confirm', SubmitType::class, ['label' => 'Confirmer la suppression'])
+            ->getForm();
+    }
+
+    #[Route('/user-delete/{id}', name: 'user_delete', methods: ['GET', 'POST'])]
+    public function delete(Request $request, EntityManagerInterface $manager, string $id): Response
+    {
+        $user = $manager->getRepository(User::class)->find($id);
+
+        if (null === $user) {
+            throw $this->createNotFoundException('L\'user n\'a pas été trouvé.');
+        }
+
+        $form = $this->createDeleteConfirmationForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $manager->remove($user);
+            $manager->flush();
+
+            $this->addFlash('success', 'L\'user a été supprimé avec succès.');
+
+            return $this->redirectToRoute('app_admin');
+        }
+
+        // Afficher la vue de confirmation si le formulaire n'a pas été soumis/validé
+        return $this->render('admin/delete.html.twig', [
+            'user' => $user,
+            'confirmationForm' => $form->createView(),
+        ]);
+    }
+
+    #[Route('/edit/user/{id}', name: 'edit_user')]
+public function editUser(Request $request, EntityManagerInterface $manager, UserPasswordHasherInterface $passwordHasher, User $user): Response
```

Comment:
```suggestion
public function edit(User $user): Response
```
Appelez les classes du constructeur